acquisitions/src/main/resources/thunderjet/mod-finance/features/batch-transaction-api.feature

```diff
@@ -0,0 +1,217 @@
+          "sourcePoLineId": "#(poLineId2)"
+        }
+      }
+    """
+    When method POST
+    Then status 201
+
+    Given path 'finance/encumbrances'
+    And request
+    """
+      {
+        "id": "#(encumbranceId3)",
+        "amount": 3,
+        "currency": "USD",
+        "description": "Encumbrance 3",
+        "fiscalYearId": "#(globalFiscalYearId)",
+        "source": "User",
+        "fromFundId": "#(fundId)",
+        "transactionType": "Encumbrance",
+        "encumbrance" : {
+          "initialAmountEncumbered": 3,
+          "status": "Unreleased",
+          "orderType": "One-Time",
+          "subscription": false,
+          "reEncumber": false,
+          "sourcePurchaseOrderId": "#(orderId)",
+          "sourcePoLineId": "#(poLineId3)"
+        }
+      }
+    """
+    When method POST
+    Then status 201
+
+
+  Scenario: Call the batch transaction API to create, update and delete transactions
+    Given path 'finance/transactions', encumbranceId3
+    When method GET
+    Then status 200
+    * def encumbrance3 = $
+    * set encumbrance3.encumbrance.status = "Released"
+
+    Given path 'finance/transactions/batch-all-or-nothing'
+    And request
+    """
+{
+  "transactionsToCreate": [
+    {
+      "id": "#(pendingPaymentId1)",
+      "amount": 10,
+      "currency": "USD",
+      "fiscalYearId": "#(globalFiscalYearId)",
+      "source": "Invoice",
+      "fromFundId": "#(fundId)",
+      "transactionType": "Pending payment",
+      "awaitingPayment": {
+        "encumbranceId": "#(encumbranceId1)",
+        "releaseEncumbrance": true
+      },
+      "sourceInvoiceId": "#(invoiceId)",
+      "sourceInvoiceLineId": "#(invoiceLineId1)"
+    },
+    {
+      "id": "#(pendingPaymentId2)",
+      "amount": 9,
+      "currency": "USD",
+      "fiscalYearId": "#(globalFiscalYearId)",
+      "source": "Invoice",
+      "fromFundId": "#(fundId)",
+      "transactionType": "Pending payment",
+      "awaitingPayment": {
+        "releaseEncumbrance": false
+      },
+      "sourceInvoiceId": "#(invoiceId)",
+      "sourceInvoiceLineId": "#(invoiceLineId2)"
+    },
+    {
+      "id": "#(allocationId)",
+      "amount": 7,
+      "currency": "USD",
+      "description": "To allocation",
+      "fiscalYearId": "#(globalFiscalYearId)",
+      "source": "User",
+      "toFundId": "#(fundId)",
+      "transactionType": "Allocation"
+    }
+  ],
+  "transactionsToUpdate": [
+    #(encumbrance3)
+  ],
+  "idsOfTransactionsToDelete": [
+    "#(encumbranceId2)"
+  ]
+}
+    """
+    When method POST
+    Then status 204
+
+
+  Scenario: Check resulting transactions and budget
+    Given path 'finance/transactions'
+    And param query = 'fromFundId==' + fundId
+    When method GET
+    Then status 200
+    And match $.totalRecords == 4
+
+    Given path 'finance/transactions'
+    And param query = 'toFundId==' + fundId
+    When method GET
+    Then status 200
+    And match $.totalRecords == 2
```

Comment:
can yo add the comment what is the second trnascation, the first one I see that it was Allocation

Reply:
added a comment